### DISCUSSION TITLE
Managing Recorded Videos or Setting Storage Limits (with MinIO and rclone)

### DISCUSSION THREAD

**Discussion by CheSauRie** [bluenviron/mediamtx#4027] [Questions]
### Question

Hello MediaMTX Community,

I'm using MediaMTX for video streaming and recording. In my setup, I also utilize MinIO for object storage and rclone to manage file uploads. I have some questions regarding managing the storage of recorded videos:

Does MediaMTX have a built-in feature to:

- Manage or limit the number of recorded videos stored locally or on a connected storage (e.g., MinIO)?
- Set a maximum storage limit (e.g., 10GB) and automatically delete the oldest recordings when the limit is reached?

With MinIO and rclone, are there any recommended strategies or best practices for:

- Automatically managing video retention policies (e.g., deleting old files or rotating storage)?
- Monitoring and enforcing storage limits for uploaded recordings?

**Possible Use Case**

Here’s what I aim to achieve:

- Manage both local storage (for temporary files) and MinIO storage efficiently.
- Automatically clean up old recordings to avoid exceeding storage limits.
- Integrate this cleanup process with my rclone workflows for file uploads.

**Environment**

MediaMTX version: [v1.9.3]
Deployment: [Docker, Kubernetes]
Object Storage: MinIO
File Management: rclone
OS: [Ubuntu 20.04]

If MediaMTX does not natively support these features, I'd appreciate any suggestions for external tools, scripts, or workflows to handle this.

Thank you in advance for your insights and help! 😊

**Comment by piyushpranjal03** [comment]
Hi CheSauRie,

> Does MediaMTX have a built-in feature to manage or limit the number of recorded videos stored locally?

Yes. Using these two properties you can achieve it.  
[recordSegmentDuration: 1h, recordDeleteAfter: 24h](https://github.com/bluenviron/mediamtx/blob/main/mediamtx.yml#L472)

--- 
  
> Does MediaMTX have a built-in feature to limit the number of recorded videos stored on MinIO or S3?

No. MediaMTX can't controll these Saas.
**But**, they (MinIO & S3) have this feature, you can define lifecycle of the file.
E.g. After how many days you want to delete the file.

---

> With MinIO and rclone, are there any recommended strategies or best practices for:
> - Automatically managing video retention policies (e.g., deleting old files or rotating storage)?
> - Monitoring and enforcing storage limits for uploaded recordings?

Video retention can be managed by MinIO Lifecycle policies.  
There is a feature called "Observability" in MinIO which will help you to monitor your files.  

---

Just a suggestion, if there is any specific reason to use MinIO then use it otherwise use S3 directly.
